flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/table/MongoDBConnectorFullChangelogDeserializationSchema.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2023 Ververica Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.connectors.mongodb.table;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.types.RowKind;
+import org.apache.flink.util.Collector;
+
+import com.mongodb.client.model.changestream.OperationType;
+import com.ververica.cdc.connectors.mongodb.internal.MongoDBEnvelope;
+import com.ververica.cdc.debezium.table.MetadataConverter;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.bson.BsonDocument;
+
+import java.time.ZoneId;
+
+/**
+ * Deserialization schema from Mongodb ChangeStreamDocument to Flink Table/SQL internal data that
+ * produces full changelog mode structure {@link RowData}.
+ */
+@PublicEvolving
+public class MongoDBConnectorFullChangelogDeserializationSchema
```

Comment:
This is a feature with public API change, I'm not sure we can pick this or not, I tend to skip this one

Reply:
Thanks @leonardBang, it makes sense to me.
The supports of MongoDB 6.0 will indeed affect the API, but it will maintain compatibility with the old version under the default configuration. I tend to skip this one as well.

Reply:
cc @GOODBOY008.